net/lighttpd/Makefile

```diff
@@ -52,41 +52,64 @@ config LIGHTTPD_SSL
 	default y
 	help
 	  Implements SSL support in lighttpd (using libopenssl). This
-	  option is required if you enable the SSL engine in your
-	  lighttpd confguration file.
+	  option is enabled by default for backwards compatibility.
+          Select one TLS module below if you enable the SSL engine in
+	  your lighttpd configuration file.
+          (mod_gnutls, mod_mbedtls, mod_nss, mod_openssl, mod_wolfssl)
 endef
 
+ifneq ($(strip $(CONFIG_LIGHTTPD_SSL)),)
+  ifeq ($(CONFIG_PACKAGE_lighttpd-mod-openssl),)
+  ifeq ($(CONFIG_PACKAGE_lighttpd-mod-mbedtls),)
+  ifeq ($(CONFIG_PACKAGE_lighttpd-mod-wolfssl),)
+  ifeq ($(CONFIG_PACKAGE_lighttpd-mod-gnutls),)
+  ifeq ($(CONFIG_PACKAGE_lighttpd-mod-nss),)
+    CONFIG_PACKAGE_lighttpd-mod-openssl=m
+  endif
+  endif
+  endif
+  endif
+  endif
+endif
+
+ifneq ($(CONFIG_PACKAGE_lighttpd-mod-auth),)
+  ifeq ($(CONFIG_PACKAGE_lighttpd-mod-authn_file),)
+    CONFIG_PACKAGE_lighttpd-mod-authn_file=m
+  endif
+endif
+
+# (lighttpd supports -Dwith_brotli if openwrt provided libbrotli as a package)
+
 MESON_ARGS += \
 	-Dwith_bzip=false \
-	-Dwith_dbi=$(if $(CONFIG_PACKAGE_lighttpd-mod-vhostdb_dbi),true,false) \
+	-Dwith_dbi=$(if $(CONFIG_PACKAGE_lighttpd-mod-authn_dbi)$(CONFIG_PACKAGE_lighttpd-mod-vhostdb_dbi),true,false) \
 	-Dwith_fam=false \
 	-Dwith_gdbm=$(if $(CONFIG_PACKAGE_lighttpd-mod-trigger_b4_dl),true,false) \
-	-Dwith_geoip=$(if $(CONFIG_PACKAGE_lighttpd-mod-geoip),true,false) \
```

Comment:
Why remove this? Just set it to false if it's no longer used.

Reply:
GeoIP version 1 database from MaxMind are deprecated and will no longer be supported as of May 2022.
GeoIP2 (version 2) databases from MaxMind are supported and used by mod_maxminddb.

Since mod_geoip has been disabled (forever?) in OpenWrt and the GeoIP version 1 databases are not available, I figured that I would remove this for cleanliness.  I'll change this to set it to `false`